### PR TITLE
Add more icons

### PR DESCRIPTION
**Pull request by wkaelj**
More icon support
 - CMake
 - Make
 - C/C++ header files
 - Cargo files
 - Git files (.gitignore, .gitmodules)

### CODE REVIEW

File: lapce-ui/src/svg.rs [hunk 1]
```diff
@@ -49,15 +49,33 @@ pub fn get_svg(name: &'static str) -> Option<Svg> {
...
-    let file_type = if path.file_name().and_then(OsStr::to_str) == Some("LICENSE") {
+    
+    // catch if path.to_str() is Some without calling it twice
+    let path_str = path.to_str();
+    if path_str == None {
+        panic!("Missing path");
+    }
+    let file_name = path_str.unwrap(); // unwrap path_str safely now
+
+    let file_type = if file_name == "LICENSE" {
```
Comment (wkaelj) [inline lapce-ui/src/svg.rs:60]: Should be file_name.contains("LICENSE")

**Review by Shasta84 — APPROVED** [review]

File: lapce-ui/src/svg.rs [hunk 2]
```diff
@@ -49,15 +49,33 @@ pub fn get_svg(name: &'static str) -> Option<Svg> {
...
+    // catch if path.to_str() is Some without calling it twice
+    let path_str = path.to_str();
+    if path_str == None {
+        panic!("Missing path");
+    }
+    let file_name = path_str.unwrap(); // unwrap path_str safely now
+
+    let file_type = if file_name.contains("LICENSE") {
         "file_type_license.svg"
+    } else if file_name.to_lowercase().contains("makefile") {
```
Comment (bugadani) [inline lapce-ui/src/svg.rs:62]: If you're still willing to work on this PR, please extract the `.to_lowercase()` conversion. In fact, if you could use some sort of case insensitive matching here, that would be best. Thanks :)

File: lapce-ui/src/svg.rs [hunk 3]
```diff
@@ -49,15 +49,33 @@ pub fn get_svg(name: &'static str) -> Option<Svg> {
 }
 
 pub fn file_svg(path: &Path) -> Svg {
-    let file_type = if path.file_name().and_then(OsStr::to_str) == Some("LICENSE") {
+    
+    // catch if path.to_str() is Some without calling it twice
+    let path_str = path.to_str();
+    if path_str == None {
```
Comment (bugadani) [inline lapce-ui/src/svg.rs:55]: This isn't really necessary. `path_str.expect("Missing path");` should be enough in place of the unwrap below.

File: lapce-ui/src/svg.rs [hunk 4]
```diff
@@ -71,12 +89,15 @@ pub fn file_svg(path: &Path) -> Svg {
...
+                    (&["svg"], "file_type_svg.svg"),
+                    (&["cmake"], "file_type_cmake.svg"),
+                    (&["make"], "file_type_make.svg"),
                 ];
 
                 for (exts, file_type) in TYPES {
                     for ext in exts.iter() {
                         if extension.eq_ignore_ascii_case(ext) {
-                            return Some(*file_type);
+                            Some(*file_type);
```
Comment (bugadani) [inline lapce-ui/src/svg.rs:100]: Why remove the return?

File: lapce-ui/src/svg.rs [hunk 5]
```diff
@@ -85,8 +106,13 @@ pub fn file_svg(path: &Path) -> Svg {
             })
             .unwrap_or("default_file.svg")
     };
-
-    get_svg(file_type).unwrap()
+    
+    let svg_file = get_svg(file_type);
+    if svg_file.is_none() {
+        panic!("Missing svg icon '{}'", file_type);
```
Comment (bugadani) [inline lapce-ui/src/svg.rs:112]: Same as above: prefer `.expect()` (or in this case, `unwrap_or_else(|| panic!())`) over an `if`

File: lapce-ui/src/svg.rs (hunk 4 above)
Comment (wkaelj) [reply lapce-ui/src/svg.rs:100]: I got a issue about it when I submit the pr the first time. Something about it not following the style guide.

File: lapce-ui/src/svg.rs (hunk 2 above)
Comment (wkaelj) [reply lapce-ui/src/svg.rs]: changed it to .eq_ignore_ascii_case

**Review by panekj — CHANGES_REQUESTED** [review]
I'm still against adding technology specific icons due to licence constraint

**Comment by bugadani** [comment]
Good point. Also I would prefer landing #768 first, if ever.

**Comment by wkaelj** [comment]
Icons not wanted

**Pull request closed without merging**